docs/source/python/timestamps.rst

```diff
@@ -0,0 +1,174 @@
+.. to you under the Apache License, Version 2.0 (the
+.. "License"); you may not use this file except in compliance
+.. with the License.  You may obtain a copy of the License at
+
+..   http://www.apache.org/licenses/LICENSE-2.0
+
+.. Unless required by applicable law or agreed to in writing,
+.. software distributed under the License is distributed on an
+.. "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+.. KIND, either express or implied.  See the License for the
+.. specific language governing permissions and limitations
+.. under the License.
+
+**********
+Timestamps
+**********
+
+Arrow/Pandas Timestamps
+=======================
+
+Arrow timestamps are stored as a 64-bit integer with column metadata to
+associate a time unit (e.g. milliseconds, microseconds, or nanoseconds), and an
+optional time zone.  Pandas (`Timestamp`) has essentially the same
+representations as Arrow.  Pandas's default time unit is nanoseconds.
+Python/Pandas timestamp types without a associated time zone are referred to as
+"Naive".  Python/Pandas timestamp types with an associated time zone are
+referred to as "Aware".   
+
+
+Timestamp Conversions
+=====================
+
+Pandas/Arrow ⇄ Spark
+--------------------
+
+Spark stores timestamps as 64-bit integers representing microseconds since
+the UNIX epoch.  It does not store any metadata about time zones with its
+timestamps.  
+
+Spark interprets timestamps with the *session local time zone*, (i.e.
+``spark.sql.session.timeZone``). If that time zone is undefined, Spark turns to
+the default system time zone. For simplicity's sake below, the session
+local time zone is always defined.
+
+This implies a few things when round-tripping timestamps:
+
+#.  Timezone information is lost.
+#.  Timestamps are truncated to microseconds.
+#.  The session time zone might have unintuitive impacts on 
+    translation of timestamp values. 
+
+Spark to Pandas (through Apache Arrow)
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+The following cases assume the Spark configuration
+``spark.sql.execution.arrow.enabled`` is set to ``"true"``.
+
+::
+
+    >>> pdf = pd.DataFrame({'naive': [datetime(2019, 1, 1, 0)], 
+    ...                     'aware': [Timestamp(year=2019, month=1, day=1, 
+    ...                               nanosecond=500, tz=timezone(timedelta(hours=-8)))]})
+    >>> pdf
+           naive                               aware
+           0 2018-10-01 2018-10-01 00:00:00.000000500-08:00
+
+    >>> spark.conf.set("spark.sql.session.timeZone", "UTC")
+    >>> utc_df = sqlContext.createDataFrame(pdf).
+    >>> utf_df.show()
+    +-------------------+-------------------+
+    |              naive|              aware|
+    +-------------------+-------------------+
+    |2019-01-01 00:00:00|2019-01-01 08:00:00|
+    +-------------------+-------------------+
+                    
+Note that conversion of the aware timestamp is shifted to reflect the time
+assuming UTC (it represents the same instant in time). 
```

Comment:
Spark treats naive timestamps as being in the system local time zone, maybe include a note that the conversion goes from system local time to UTC.

Reply:
Done